dpnp/backend_bitwise.pyx

```diff
@@ -47,54 +47,23 @@ __all__ += [
 
 
 cpdef dparray dpnp_bitwise_and(dparray array1, dparray array2):
-    cdef dparray result = dparray(array1.shape, dtype=array1.dtype)
-
-    for i in range(result.size):
-        result[i] = array1[i] & array2[i]
-
-    return result
+    return call_fptr_2in_1out(DPNP_FN_BITWISE_AND, array1, array2, array1.shape)
 
 
 cpdef dparray dpnp_bitwise_or(dparray array1, dparray array2):
-    cdef dparray result = dparray(array1.shape, dtype=array1.dtype)
-
-    for i in range(result.size):
-        result[i] = array1[i] | array2[i]
-
-    return result
+    return call_fptr_2in_1out(DPNP_FN_BITWISE_OR, array1, array2, array1.shape)
 
 
 cpdef dparray dpnp_bitwise_xor(dparray array1, dparray array2):
-    cdef dparray result = dparray(array1.shape, dtype=array1.dtype)
-
-    for i in range(result.size):
-        result[i] = array1[i] ^ array2[i]
-
-    return result
+    return call_fptr_2in_1out(DPNP_FN_BITWISE_XOR, array1, array2, array1.shape)
 
 
 cpdef dparray dpnp_invert(dparray arr):
-    cdef dparray result = dparray(arr.shape, dtype=arr.dtype)
-
-    for i in range(result.size):
-        result[i] = ~arr[i]
-
-    return result
+    return call_fptr_1in_1out(DPNP_FN_INVERT, arr, arr.shape)
 
 
 cpdef dparray dpnp_left_shift(dparray array1, dparray array2):
-    cdef dparray result = dparray(array1.shape, dtype=array1.dtype)
-
-    for i in range(result.size):
-        result[i] = array1[i] << array2[i]
-
-    return result
-
+    return call_fptr_2in_1out(DPNP_FN_LEFT_SHIFT, array1, array2, array1.shape)
```

Comment:
Where result array memory is created?

Reply:
The result array is created inside of `call_fptr_2in_1out`:
https://github.com/IntelPython/dpnp/blob/07bee6d87cd613224059ff30ce4d2511a89f69c3/dpnp/backend.pyx#L284.